views/install.sh.erb

```diff
@@ -490,8 +492,36 @@ install_file() {
 
 echo "Downloading Chef $version for ${platform}..."
 
+# Create metadata filename
+metadata_filename="metadata"
+
+if test "x$project" != "xchef"; then
+  metadata_filename="${metadata_filename}-$project"
+fi
+
+if test "x$version" = "x"; then
+  metadata_filename="${metadata_filename}-latest"
+else
+  metadata_filename="${metadata_filename}-${version}"
+fi
+
+metadata_filename="${metadata_filename}-${platform}-${platform_version}-${machine}"
 
-metadata_filename="$tmp_dir/metadata.txt"
+if test "x$prerelease" = "xtrue"; then
+  metadata_filename="${metadata_filename}-preleases"
+fi
+if test "x$nightlies" = "xtrue"; then
+  metadata_filename="${metadata_filename}-nightlies"
+fi
+
+metadata_filename="${metadata_filename}.txt"
+
+# Prepend path to metadata filename
+if test "x$cmdline_dl_dir" != "x"; then
+  metadata_filename="$cmdline_dl_dir/$metadata_filename"
```

Comment:
That is the value passed in via the `-d` argument.  So `sudo bash install.sh -d /tmp` would have `$cmdline_dl_dir` set to `/tmp`.

It's used near the bottom as the caching directory for the package download, [line 593](https://github.com/ben-rosio/opscode-omnitruck/blob/feature/offline-install/views/install.sh.erb#L593).

So if `-d` is set than it'll download the metadata file there, if it isn't than it'll download it to the `$tmp_dir`, and delete it after the script completes successfully.